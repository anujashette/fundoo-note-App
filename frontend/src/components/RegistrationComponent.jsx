import React, { Component } from 'react';
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent';
import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator';
import TextField from '@material-ui/core/TextField'
import Snackbar from '@material-ui/core/Snackbar'
import { register } from '../services/UserService';
// import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import '../App.css'

class RegistrationComponent extends Component {
    constructor(props) {
        super(props);
        this.state = {
            formData: {
                email: '',
                firstname: '',
                lastname: '',
                password: '',
            },
            snackbaropen: false,
            snackbarmsg: '',
        }
        this.handleSubmit = this.handleSubmit.bind(this)
        this.handleLogin = this.handleLogin.bind(this)
    }

    snackBarClose = () => {
        this.setState({ snackbaropen: false })
    }
    handleChange = (event) => {
        const { formData } = this.state;
        formData[event.target.name] = event.target.value;
        this.setState({ formData });
    }

    handleSubmit(event) {
        event.preventDefault();
        console.log('form data in submit', this.state.formData)

        if (!this.state.formData.email) {
            console.log('email should not be empty')
            this.setState({ snackbaropen: true, snackbarmsg: 'Email should not be empty' })
        }
        else if (!this.state.formData.firstname) {
            console.log('firstname should not be empty')
            this.setState({ snackbaropen: true, snackbarmsg: 'Firstname should not be empty' })
        }
        else if (!this.state.formData.lastname) {
            console.log('lastname should not be empty')
            this.setState({ snackbaropen: true, snackbarmsg: 'Lastname should not be empty' })
        }
        else if (!this.state.formData.password) {
            console.log('Password should not be empty')
            this.setState({ snackbaropen: true, snackbarmsg: 'Password should not be empty' })
        }
        else if (this.state.formData.password.length < 6) {
            this.setState({ snackbaropen: true, snackbarmsg: 'password too short. At least 6 characters' })
        }
        else {
            let userInput = {
                email: this.state.formData.email,
                firstname: this.state.formData.firstname,
                lastname: this.state.formData.lastname,
                password: this.state.formData.password
            }

            // event.target.reset();
            // this.setState({
            //     firstname: '',
            //     lastname: ''
            // })
            var formData = Object.assign({}, this.state.formData);
            Object.keys(formData).map((key, index) => {
                formData[key] = '';
                return ''
            });

            this.setState({ formData })

            register(userInput)
                .then((response) => {
                    console.log('register', response.data)
                    this.setState({ snackbaropen: true, snackbarmsg: response.data.message })
                    this.setState({
                        firstname: '',
                        lastname: ''
                    })
                    var formData = Object.assign({}, this.state.formData);
                    Object.keys(formData).map((key, index) => {
                        formData[key] = '';
                        return ''
                    });
                }).catch((errorMessages) => {
                    console.log('error', errorMessages)
                    this.setState({ snackbaropen: true, snackbarmsg: 'User already exist' })
                })
        }
    }

    handleLogin = () => {
        console.log('in handle Login')
        this.props.props.history.push('/')
    }

    render() {
        return (
            <div>
                <Snackbar
                    anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
                    open={this.state.snackbaropen}
                    autoHideDuration={4000}
                    onClose={this.snackBarClose}

                    message={<span id="message-id">{this.state.snackbarmsg}</span>}
                ></Snackbar>

                <Card id="cardid" >
                    <CardContent>
                        <Typography variant="h5">
                            {/* FundooNotes */}
                            <span className="font-color">F</span>
                            <span className="font-color">u</span>
                            <span className="font-color">n</span>
                            <span className="font-color">d</span>
                            <span className="font-color">o</span>
                            <span className="font-color">o</span>
                        </Typography>
                        <Typography variant="h5" component="h2">
                            Create your Fundoo Account
                        </Typography>

                        <br />

                        <ValidatorForm
                            autoComplete="off"
                            ref="form"
                            onSubmit={this.handleSubmit}
                        >
                            <div className="NameText">
                                <TextField
                                    required
                                    label="First name"
                                    onChange={this.handleChange}
                                    name="firstname"
                                    value={this.state.formData.firstname}
                                    className="FName"
                                />

                                <TextField
                                    required
                                    label="Last name"
                                    onChange={this.handleChange}
                                    name="lastname"
                                    value={this.state.formData.lastname}
                                    className="FName"
                                />
                            </div>

                            <br />
                            <TextValidator
                                label="Email"
                                onChange={this.handleChange}
                                name="email"
                                value={this.state.formData.email}
                                className="Email"
                                validators={['required', 'isEmail']}
                                errorMessages={['this field is required', 'email is not valid']}
                            />

                            <br />
                            <br />

                            <TextValidator
                                label="Password"
                                onChange={this.handleChange}
                                name="password"
                                type='password'
                                value={this.state.formData.password}
                                validators={['required']}
                                className="Password"
                                errorMessages={['this field is required']}
                            />
                            <br />
                            {/* <label className="label-font">Use 6 or more characters with letters, numbers & symbols</label> */}
                            <br />

                            <div className="Button">
                                <button
                                    className="login-button-letter"
                                    onClick={this.handleLogin}>
                                    Sign in instead
                                </button>
                                <button
                                    onClick={this.handleSubmit}
                                    type="submit"
                                    className="button-letter">
                                    Signup
                                </button>
                            </div>
                        </ValidatorForm>
                    </CardContent>
                </Card>
            </div>
        );
    }
}

export default RegistrationComponent;