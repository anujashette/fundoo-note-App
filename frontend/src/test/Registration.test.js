import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import enzyme, { shallow, configure, mount } from 'enzyme';
import RegistrationComponent from '../components/RegistrationComponent';
configure({ adapter: new Adapter() });


describe("LoginComponent", () => {
    it("should render LoginComponent", () => {
        const wrapper = shallow(<RegistrationComponent />);
        expect(wrapper.exists()).toBe(true)
    });

    it("Should validated input fields", () => {
        const wrapper = shallow(<RegistrationComponent />)
        let prevented = false
        wrapper.find("ValidatorForm").simulate("submit", {
            preventDefault: () => {
                prevented = true
            }
        });
        expect(prevented).toBe(true)

    });

    it("Should validated input fields", () => {
        const wrapper = shallow(<RegistrationComponent />)
        expect(wrapper.getElements()).toMatchSnapshot();
    });
    
    // it("onChange param is the same value as the input element's value property", () => {
    //     const wrapper = shallow(<RegistrationComponent />)

    //     wrapper.find('TextValidator').simulate('click', { target: { value: 'matched' } });
    //     expect('email').toBe('matched');
    // });
});