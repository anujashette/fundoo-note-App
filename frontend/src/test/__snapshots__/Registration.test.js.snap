// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent Should validated input fields 1`] = `
Array [
  <div>
    <ForwardRef(WithStyles)
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
      autoHideDuration={4000}
      message={
        <span
          id="message-id"
        >
          
        </span>
      }
      onClose={[Function]}
      open={false}
    />
    <ForwardRef(WithStyles)
      id="cardid"
    >
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          variant="h5"
        >
          <span
            className="font-color"
          >
            F
          </span>
          <span
            className="font-color"
          >
            u
          </span>
          <span
            className="font-color"
          >
            n
          </span>
          <span
            className="font-color"
          >
            d
          </span>
          <span
            className="font-color"
          >
            o
          </span>
          <span
            className="font-color"
          >
            o
          </span>
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          component="h2"
          variant="h5"
        >
          Create your Fundoo Account
        </ForwardRef(WithStyles)>
        <br />
        <ValidatorForm
          autoComplete="off"
          debounceTime={0}
          onError={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="NameText"
          >
            <ForwardRef(WithStyles)
              className="FName"
              label="First name"
              name="firstname"
              onChange={[Function]}
              required={true}
              value=""
            />
            <ForwardRef(WithStyles)
              className="FName"
              label="Last name"
              name="lastname"
              onChange={[Function]}
              required={true}
              value=""
            />
          </div>
          <br />
          <TextValidator
            className="Email"
            errorMessages={
              Array [
                "this field is required",
                "email is not valid",
              ]
            }
            label="Email"
            name="email"
            onChange={[Function]}
            validatorListener={[Function]}
            validators={
              Array [
                "required",
                "isEmail",
              ]
            }
            value=""
          />
          <br />
          <br />
          <TextValidator
            className="Password"
            errorMessages={
              Array [
                "this field is required",
              ]
            }
            label="Password"
            name="password"
            onChange={[Function]}
            type="password"
            validatorListener={[Function]}
            validators={
              Array [
                "required",
              ]
            }
            value=""
          />
          <br />
          <br />
          <div
            className="Button"
          >
            <button
              className="login-button-letter"
              onClick={[Function]}
            >
              Sign in instead
            </button>
            <button
              className="button-letter"
              onClick={[Function]}
              type="submit"
            >
              Signup
            </button>
          </div>
        </ValidatorForm>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </div>,
]
`;
